Add Chainguard Libraries for JavaScript docs (#2305)
chainguard-dev/edu: content/chainguard/libraries/javascript/overview.md

---
title: "Chainguard Libraries for JavaScript Overview"
linktitle: "JavaScript Overview"
description: "JavaScript libraries for your application development"
type: "article"
date: 2025-06-05T09:00:00+00:00
lastmod: 2025-06-05T09:00:00+00:00
draft: false
tags: ["Chainguard Libraries", "JavaScript", "Overview"]
menu:
  docs:
    parent: "javascript"
weight: 051
toc: true
---

> ## Introduction

Author: oops .. fixed

> **Chainguard Libraries for JavaScript** represents a major ecosystem support
> with [Chainguard Libraries](/chainguard/libraries/overview/). The JavaScript

Reviewer: ```suggestion
**Chainguard Libraries for JavaScript** is a major ecosystem supported
by [Chainguard Libraries](/chainguard/libraries/overview/). The JavaScript
```

ecosystem consists of thousands of open source projects from the communities
around [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript),
[TypeScript](https://www.typescriptlang.org/), [Node.js](https://nodejs.org/),
[React](https://react.dev/), [Vue.js](https://vuejs.org/),
[Angular](https://angular.io/), [Svelte](https://svelte.dev/),

> [Next.js](https://nextjs.org/), and [Express](https://expressjs.com/) and many

Author: fixed

others.

The main public repository for JavaScript packages is the [npm
Registry](https://npmjs.com/). Launched in 2010, the npm Registry has grown to
become the largest software registry in the world, hosting over two million
packages. It serves as the central hub for open source JavaScript libraries,
tools, and frameworks, supporting a vibrant and rapidly evolving ecosystem. The
registry is widely used by developers for both client-side and server-side
JavaScript projects, and its scale and history make it a critical resource for
modern application development.

It is the default repository in all commonly used build tools from the
JavaScript community, including [npm](https://www.npmjs.com/),
[pnpm](https://pnpm.io/), [Yarn](https://classic.yarnpkg.com/), and [Yarn
Berry](https://yarnpkg.com/), and uses the npm repository format. Chainguard
Libraries for JavaScript covers all open source artifacts from the npm Registry.

Chainguard Libraries for JavaScript provides access to all open source libraries

> commonly used. New releases of common packages requested by customer builds are

Author: Nice .. reworded

added to the growing index by an automated system.

You can use Chainguard Libraries for JavaScript alongside third-party software
repositories to create a single source of truth with your repository manager
application. 

## Runtime requirements

The runtime requirements for JavaScript packages available from Chainguard
Libraries for JavaScript are identical to the requirements of the original
upstream project. For example, if a package retrieved from the npm Registry
requires Node.JS v22 or higher, the same Node.JS v22 requirement applies to the

> package from Chainguard Libraries for Java. Same details apply to JavaScript or

Reviewer 2: This last sentence reads funny to me. Maybe I'm misunderstanding what it's saying but here's my swing at it:

```
The same applies to JavaScript, Typescript, or React versions, as well as any other requirements of the original upstream project.
```

Author: Nice .. using that.

TypeScript, React versions, and any other requirements.

## Technical details

The [username and password retrieved with
chainctl](/chainguard/libraries/access/) are required to access the Chainguard
Libraries for Java repository. The URL for the repository is:

```
https://libraries.cgr.dev/javascript/
```

This Chainguard Libraries for JavaScript repository uses the npm repository
format and only includes release artifacts of the libraries built by Chainguard
from source.

The URL does not expose a browsable directory structure.

The Chainguard Libraries for JavaScript registry does not include all packages
from the npm registry.

> 

Reviewer 2: There are a lot of one sentence paragraphs in this section. I'd combine some of them to give the text more rhythm. Like these two

Author: Good point .. reordered it all and cleaned up

Specifically, the following components are not included:

* Packages without any source code available, including malware as well as
  proprietary packages.

> * Packages that use post-install script.

Reviewer: ```suggestion
* Packages that use post-install scripts.
```

* Packages that are flagged as malware during our build process.

As a result, you must configure the repository as the first point of contact and
request for any retrieval of a package. This ensures that any library that is
available from Chainguard is also used. In addition, any failed requests are
flagged at Chainguard and backfill processes are run where possible.

> At the same time, you must might need to continue to use other repositories that

Reviewer 2: Also, I want to make this paragraph more active, there's a lot of passive voice in it now:

```
As a result, you must configure the repository as the first point of contact for all package retrievals. This setup directs requests to Chainguard, ensuring that all available libraries are used. If a request fails, Chainguard flags it and runs backfill processes where possible.
```

> At the same time, you must might need to continue to use other repositories that

Author: Great.. thanks .. switched over.
